.github/workflows/release.yaml

```diff
@@ -0,0 +1,33 @@
+# SPDX-License-Identifier: MIT
+
+name: release
+
+on:
+  push:
+    # run only against tags
+    tags:
+      - 'v*'
+
+permissions:
+  contents: write
+
+jobs:
+  goreleaser:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - run: git fetch --force --tags
+      - uses: actions/setup-go@v5
+        with:
+          go-version: '1.23.1'
+
+      - name: Run GoReleaser
+        uses: goreleaser/goreleaser-action@v6
```

Comment:
Suggestion to pin the versions for security concerns. Refer to other workflow files, i.e. https://github.com/etcd-io/auger/blob/8fe6cb4019f8c564197003da3573f6c2f8c46ad7/.github/workflows/static-analysis.yaml#L9-L10